Observer/NonBoltOrderObserver.php

```diff
@@ -0,0 +1,201 @@
+        MetricsClient $metricsClient
+    ) {
+        $this->apiHelper = $apiHelper;
+        $this->bugsnag = $bugsnag;
+        $this->cartHelper = $cartHelper;
+        $this->configHelper = $configHelper;
+        $this->dataObjectFactory = $dataObjectFactory;
+        $this->logHelper = $logHelper;
+        $this->metricsClient = $metricsClient;
+    }
+
+    /**
+     * @param Observer $observer
+     */
+    public function execute(Observer $observer)
+    {
+        try {
+            $order = $observer->getEvent()->getOrder();
+            if (!$order) {
+                // for multi-address orders, the event will have a list of orders instead
+                $orders = $observer->getEvent()->getOrders();
+                if (!$orders || count($orders) < 1) {
+                    $this->metricsClient->processCountMetric("non_bolt_order_creation.no_order", 1);
+                    return;
+                }
+
+                $order = $orders[0];
+            }
+
+            $payment = $order->getPayment();
+            if ($payment && $payment->getMethod() == Payment::METHOD_CODE) {
+                return;
+            }
+
+            $quote = $observer->getEvent()->getQuote();
+            if (!$quote) {
+                $this->metricsClient->processCountMetric("non_bolt_order_creation.no_quote", 1);
+                return;
+            }
+
+            $items = $quote->getAllVisibleItems();
+            if (!$items) {
+                $this->metricsClient->processCountMetric("non_bolt_order_creation.no_items", 1);
+                return;
+            }
+
+            $cart = $this->cartHelper->buildCartFromQuote($quote, $quote, $items, null, true);
+            $customer = $quote->getCustomer();
+            $storeId = $order->getStoreId();
+            $result = $this->createNonBoltOrder($cart, $customer, $storeId);
+            if ($result != 200) {
+                $this->metricsClient->processCountMetric("non_bolt_order_creation.failure", 1);
+                return;
+            }
+        } catch (Exception $exception) {
+            $this->metricsClient->processCountMetric("non_bolt_order_creation.failure", 1);
+            $this->bugsnag->notifyException($exception);
+            return;
+        } catch (Error $error) {
+            // catch errors so failures here don't prevent orders from being created
+            $this->metricsClient->processCountMetric("non_bolt_order_creation.failure", 1);
+            $this->bugsnag->notifyException($error);
+            return;
+        }
+
+        $this->metricsClient->processCountMetric("non_bolt_order_creation.success", 1);
+    }
+
+    /**
+     * Call Bolt Create Non-Bolt Order API
+     *
+     * @param array $cart
+     * @param CustomerInterface $customer
+     * @param int $storeId
+     * @return Response|int
+     * @throws LocalizedException
+     * @throws Zend_Http_Client_Exception
+     */
+    protected function createNonBoltOrder($cart, $customer, $storeId)
```

Comment:
Does the `$customer` always have associated email and phone data? First and last names also. What happens in case of a guest checkout? As a fallback, if needed, this data can be collected from the quote and / or quoteAddress. Please investigate.

Reply:
In all of my testing, the `$customer` had the necessary information. If it comes up, I can always add that later.